File: frontEnd/intermediateAlgorithm/sumAllNumsInRange.js
/*jslint node: true */
'use strict';

function sumAll (arr) {
  var max = Math.max(...arr);
  var min = Math.min(...arr);
  for (var i = 0; i <= max - min; i++) {
    arr[i] = min + i;
  }
  var sum = arr.reduce(function (a, b) {
    return a + b;
  });
  return sum;
}

console.log(sumAll([1, 4]));
console.log(sumAll([10, 5]));

File: oldCurriculum/goCode/intermediateGo/sumAllNumsInRange.go
package main

import (
	"fmt"
)

func sumAll(arr [2]int) int {
	var sum int
	min := arr[0]
	max := arr[1]
	if min > max {
		aux := min
		min = max
		max = aux
	}
	for i := min; i <= max; i++ {
		sum += i
	}
	return sum
}

func sumAllNumsIntRange() {
	fmt.Println(sumAll([2]int{1, 4}))
	fmt.Println(sumAll([2]int{4, 1}))
	fmt.Println(sumAll([2]int{5, 10}))
	fmt.Println(sumAll([2]int{10, 5}))
}
